lib/libPlugin.lua

```diff
@@ -0,0 +1,724 @@
+  label = "upscale",
+  value = 1, "no", "yes",
+}
+
+libPlugin.global = dt.new_widget("box"){
+  orientation = "vertical",
+  dt.new_widget("label"){ label = "Global Options" },
+  dt.new_widget("separator"){},
+  dt.new_widget("label"){ label = "Max Size" },
+  libPlugin.height_box,
+  libPlugin.width_box,
+  libPlugin.upscale,
+}
+
+-- execute button
+-- TODO: save the values from the various widgets when process is pressed
+
+libPlugin.button = dt.new_widget("button"){
+  label = "Process",
+  clicked_callback = function (_)
+
+    -- get the gui parameters
+    local export_format = libPlugin.format_combobox.value
+
+    -- build the image table
+    local img_table, cnt = libPlugin.build_image_table(dt.gui.action_images, export_format)
+    log.msg(log.debug, "image count is " .. cnt)
+
+    -- make sure there is enough images
+    if plugins[libPlugin.processor_combobox.value].DtPluginMinImages <= cnt then
+
+      -- export the images
+      local success = libPlugin.do_export(img_table, export_format, libPlugin.height.text, libPlugin.width.text, libPlugin.upscale.value)
+
+      -- call the processor
+      log.msg(log.debug, processor_cmds[libPlugin.processor_combobox.value])
+      processor_cmds[libPlugin.processor_combobox.value](img_table, plugins[libPlugin.processor_combobox.value])
+    else
+      log.msg(log.error, "Insufficient images selected, " .. plugins[libPlugin.processor_combobox.value].DtPluginMinImages .. " required")
+    end
+  end
+}
+
+--[[
+  NAME
+    libPlugin.register_processor_lib - build and install the external processor gui
+
+  SYNOPSIS
+    result = dtutils.split_filepath(path_string)
+      filepath - path and filename
+
+  DESCRIPTION
+    split_filepath splits a filepath into the path, filename, basename and filetype and puts
+    that in a table
+
+  RETURN VALUE
+    result - a table containing the path, filename, basename, and filetype
+
+  ERRORS
+
+
+
+]]
+
+function libPlugin.register_processor_lib(name_table)
+
+  log.msg(log.info, "name_table length is ", #name_table)
+
+  -- since we don't know how many processors are going to be present at startup, we just put
+  -- a placeholder in the combobox and load the correct values later
+
+  libPlugin.processor_combobox = dt.new_widget("combobox"){
+    label = "processor",
+    tooltip = "pick a processor",
+--    value = 1, unpack(name_table), bug #11184
+    value = 1, "placeholder",
+    changed_callback = function(_)
+
+      -- reset the processor widget
+      libPlugin.processor[4] = nil
+
+      -- set the processor widget to the appropriate one for the selected processor
+      libPlugin.processor[4] = processors[libPlugin.processor_combobox.value]
+
+      -- update the export formats to those allowed for the processor
+      local supported_formats = libPlugin.get_supported_formats(plugins[libPlugin.processor_combobox.value])
+      dtutils.updateComboboxChoices(libPlugin.format_combobox, supported_formats)
+    end
+  }
```

Comment:
here too, you might want to use a stack widget and add a child when a new processor is registered